mobilecoind-json/src/main.rs

```diff
@@ -244,9 +244,48 @@ fn request_code(
 
 #[derive(Deserialize, Serialize, Default)]
 struct JsonPublicAddress {
+    /// Hex encoded compressed ristretto bytes
     view_public_key: String,
+    /// Hex encoded compressed ristretto bytes
     spend_public_key: String,
-    fog_url: String,
+    /// Fog Report Server Url
+    fog_report_url: String,
+    /// Hex encoded signature bytes
+    fog_authority_sig: String,
+    /// String label for fog reports
+    fog_report_key: String,
+}
+
+// Helper conversion between json and protobuf
+impl TryFrom<&JsonPublicAddress> for PublicAddress {
```

Comment:
this addresses comment https://github.com/mobilecoinofficial/mobilecoin/pull/242#discussion_r446309937